jbpm-wb-kie-server/jbpm-wb-kie-server-backend/src/test/java/org/jbpm/workbench/ks/integration/KieServerQueryDefinitionLoaderTest.java

```diff
@@ -0,0 +1,79 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.workbench.ks.integration;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.enterprise.event.Event;
+
+import org.jbpm.workbench.ks.integration.event.QueryDefinitionLoaded;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.server.api.model.definition.QueryDefinition;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.MockitoAnnotations.openMocks;
+
+@RunWith(MockitoJUnitRunner.class)
+public class KieServerQueryDefinitionLoaderTest {
+
+    private final Map<String, QueryDefinition> receivedEvents = new HashMap<>();
+
+    @Mock
+    private Event<QueryDefinitionLoaded> event;
+
+    @InjectMocks
+    private KieServerQueryDefinitionLoader kieServerQueryDefinitionLoader;
+
+    @Before
+    public void setup() {
+        openMocks(this);
+    }
+
+    @Test
+    public void testJbpmProcessInstancesWithVariables() {
+        QueryDefinition query = QueryDefinition.builder()
+                .name("jbpmProcessInstancesWithVariables")
+                .expression("select vil.processInstanceId, vil.processId, vil.id, " +
+                                    "vil.variableId, vil.value from VariableInstanceLog vil " +
+                                    "left join VariableInstanceLog vil2 on vil.processInstanceId = vil2.processInstanceId " +
+                                    "and vil.variableId = vil2.variableId and vil.id < vil2.id where vil2.id is null")
+                .build();
+        testQueryDefinitionLoaded(query);
+    }
+
+    private void testQueryDefinitionLoaded(QueryDefinition queryDefinition) {
+        doAnswer(invocation -> {
+            QueryDefinitionLoaded queryDefinitionLoaded = invocation.getArgument(0);
+            receivedEvents.put(queryDefinitionLoaded.getDefinition().getName(), queryDefinitionLoaded.getDefinition());
+            return null;
+        }).when(event).fire(any(QueryDefinitionLoaded.class));
+        this.kieServerQueryDefinitionLoader.init();
+
+        assertTrue(receivedEvents.size() > 0);
```

Comment:
Is a more descriptive assertion available, e.g. using hamcrest?

Reply:
Just added some description to the assert. Let me know if that's ok for you, otherwise I can use hamcrest, but it will require a change in the pom.xml as this dep is not included there